source/_drafts/videojs-contrib-ads-6.md

```diff
@@ -0,0 +1,43 @@
+---
+title: videojs-contrib-ads 6
+author:
+  name: Greg Smith
+  github: incompl
+tags:
+---
+
+A major update to videojs-contrib-ads has been released. In this post, we'll take a look at what videojs-contrib-ads offers and what has improved in this latest version.
+
+## What is contrib-ads?
+
+Videojs-contrib-ads is a framework for creating Video.js ad plugins. Seamlessly integrating ad support into a video player can be a daunting task, especially if you have other plugins that may be effected. For example, playing ads may result in additional [media events](https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Media_events). Imagine an analytics system that listens for "loadstart" events: it would start seeing double when extra `loadstart` events result from preroll ads. contrib-ads has a feature called [Redispatch](https://github.com/videojs/videojs-contrib-ads#redispatch) that makes sure ads do not trigger extra media events. This keeps things simple as possible without breaking other plugins. [Additional benefits of using contrib-ads are listed in the readme](https://github.com/videojs/videojs-contrib-ads#benefits).
+
+## Version 6
+
+Version 6 of contrib-ads is a maintenance release that includes a major code refactor. The core of the project's behavior is handled by a state machine that advances through various states as the player plays content, preroll ads, midroll ads, and postroll ads. The initial state machine was designed to implement a specific set of functionality when the project was first conceptualized. Over the years, the project became much more fully-featured and many new scenarios and edge-cases arose. In this refactor, the state machine was updated to precisely match the modern feature-set. This has resulted in a more maintainable and reliable codebase.
+
+## How it works
+
+contrib-ads has a concept called _ad mode_. Ad mode is [strictly defined](https://github.com/videojs/videojs-contrib-ads#ad-mode-definition) as _if content playback is blocked by the ad plugin_. This does not necessarily mean that an ad is playing. For example, if there is a preroll ad, after a play request we show a loading spinner until ad playback begins. This is considered part of ad mode. The public method `player.ads.isInAdMode()` can be used to check if we are in ad mode.
```

Comment:
Thoughts on making this sentence `Ad mode is [strictly defined](https://github.com/videojs/videojs-contrib-ads#ad-mode-definition) as _if content playback is blocked by the ad plugin_.` into a pullquote?
```ejs
{% pullquote right %}
Ad mode is [strictly defined](https://github.com/videojs/videojs-contrib-ads#ad-mode-definition) as _if content playback is blocked by the ad plugin_.
{% endpullquote %}

Reply:
that is, in addition to having it inline with the paragraph.